hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -415,9 +415,7 @@ public static DataStreamSink<CompactionCommitEvent> compact(Configuration conf,
             new CompactionPlanOperator(conf))
         .setParallelism(1) // plan generate must be singleton
         .setMaxParallelism(1)
-        // make the distribution strategy deterministic to avoid concurrent modifications
-        // on the same bucket files
-        .keyBy(plan -> plan.getOperation().getFileGroupId().getFileId())
+        .rebalance()
```

Comment:
One operator one file group, rebalance is ok.
This is what we do in HoodieFlinkCompactor
https://github.com/apache/hudi/blob/b7e11b1d018e07aa85bf7de49c05bc2ccb71262d/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/HoodieFlinkCompactor.java#L289-L303

Reply:
Is `rebanlance` deterministic for multiple concurrent tasks, when there are task failover and retries, two task may has the risk of sharing a common compaction file group.

Reply:
Did you test it offline already?

Reply:
I tested on my cluster.  But I didn't test the failed and retry case.

Reply:
Offline conpaction uses rebalance.